modules/@apostrophecms/widget-type/ui/apos/mixins/AposWidgetMixin.js

```diff
@@ -61,20 +63,26 @@ export default {
         // AposAreas manager can spot any new area divs.
         // This will also run the player
         setTimeout(function() {
+          self.setPlayerEl();
           apos.bus.$emit('widget-rendered');
         }, 0);
       } catch (e) {
         this.rendered = '<p>Unable to render this widget.</p>';
         console.error('Unable to render widget. Possibly the schema has been changed and the existing widget does not pass validation.', e);
       }
     },
-    runPlayer() {
-      if (!this.playerOpts) {
-        return;
+    setPlayerEl() {
+      if (this.playerOpts) {
+        const el = this.$el.querySelector(this.playerOpts.selector);
+        if (el && this.playerOpts.player) {
+          this.playerEl = el;
+        }
       }
-      const el = this.$el.querySelector(this.playerOpts.selector);
-      if (el && this.playerOpts.player) {
-        this.playerOpts.player(el);
+    },
+    runPlayer() {
+      if (this.playerEl && !this.playerEl.played) {
```

Comment:
Yes this is the pattern — custom properties on DOM elements are allowed, and it's the same object in JS for as long as it's the same DOM element — but please namespace like we do in runPlayers with the name `aposWidgetPlayed` to avoid any conflict with future DOM properties or similar bookkeeping logic at project level.